src/vs/editor/common/modes.ts

```diff
@@ -633,6 +633,11 @@ export interface CompletionItemProvider {
 	 */
 	_debugDisplayName?: string;
 
+	/**
+	 * @internal
+	 */
+	_source?: string;
```

Comment:
This basically duplicates `_debugDisplayName` but looses precision. Extensions like TypeScript register a couple of completion item providers (I believe 3 for JS and 3 for TS) and the debug display name allows you to tell them apart (with some chance of error). Like, you won't be able to differentiate the accept rate of JSDoc completions vs semantic completions etc.

Reply:
Hm, so I had originally kept them separate because the snippet provider's `source` could be more specific than the current display name. But given that's not actually happening yet I agree this is not needed.